### PR TITLE
Remove Outer Ticker Loop - Dead Code

### PR DESCRIPTION
**Pull request by dvonthenen**
## Proposed changes

Removing dead code for v`1.4.0` as suggested here:
https://github.com/deepgram/deepgram-go-sdk/issues/244#issuecomment-2182774904

The change looks far more significant than it actually is. This deletes this section of code (https://github.com/deepgram/deepgram-go-sdk/blob/main/pkg/client/live/client.go#L301-L310) and then fixes the indentation which is why it looks like there are so many changes.

Also, removes emitting this error, for this error code, because it's actually an intended error:
https://github.com/deepgram/deepgram-go-sdk/blob/main/pkg/client/live/client.go#L346-L350

Adds a simple non-interactive example: `examples/speak/stream/simple`

Tested all examples and they appear to work as intented.

## Types of changes

What types of changes does your code introduce to the community Go SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Summary by CodeRabbit

- **New Features**
  - Improved error handling and logging for websocket message reading.
  - Added constant for invalid WebSocket message type.

- **Refactor**
  - Restructured error handling in the `listen` method for better clarity and robustness.
  - Moved constants within the codebase for better organization and maintainability.

- **Chores**
  - Introduced a new method to handle message reading, enhancing code modularity.

### CODE REVIEW

**Review by SandraRodgers — APPROVED** [review]

File: pkg/client/live/client.go [hunk 1]
```diff
@@ -298,149 +298,131 @@ func (c *Client) internalConnectWithCancel(ctx context.Context, ctxCancel contex
...
-					c.muConn.Unlock()
-
-					klog.V(3).Infof("listen: Connection is not valid\n")
-					klog.V(6).Infof("live.listen() LEAVE\n")
-					return
+		}
+
+		// release the lock
+		c.muConn.Unlock()
+
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/live/client.go:318]: `internalConnect` func only uses the mutex when calling on initial connect. the mutex is necessary here.

File: pkg/client/live/client.go [hunk 2]
```diff
@@ -298,149 +298,131 @@
...
+			if msgType == websocket.TextMessage {
+				err := c.router.Message(byMsg)
+				if err != nil {
+					klog.V(1).Infof("live.listen(): router.Message failed. Err: %v\n", err)
 				}
+			} else {
+				// this shouldn't happen, but let's log it
+				klog.V(7).Infof("live.listen(): msg recv: type %d, len: %d\n", msgType, len(byMsg))
 			}
+		} else {
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/live/client.go:424]: Will keep it this way for right now. don't want to put too much refactoring in at a time.

File: pkg/client/speak/constants.go [hunk 3]
```diff
@@ -50,3 +45,10 @@ var (
 	// ErrInvalidConnection connection is not valid
 	ErrInvalidConnection = errors.New("connection is not valid")
 )
+
+// internal constants for retry, waits, back-off, etc.
+const (
+	defaultDelayBetweenRetry int64 = 2
+
+	invalidWebSocketMsgType int = -1
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/speak/constants.go]: Removed!

File: pkg/client/live/constants.go [hunk 4]
```diff
@@ -51,3 +42,13 @@
...
 )
+
+// internal constants for retry, waits, back-off, etc.
+const (
+	flushPeriod = 500 * time.Millisecond
+	pingPeriod  = 5 * time.Second
+
+	defaultDelayBetweenRetry int64 = 2
+
+	invalidWebSocketMsgType int = -1
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/live/constants.go]: Removed!

File: pkg/client/speak/client_stream.go [hunk 5]
```diff
@@ -292,155 +292,137 @@ func (c *StreamClient) internalConnectWithCancel(ctx context.Context, ctxCancel
...
-								klog.V(1).Infof("StreamClient.listen(): router.Message failed. Err: %v\n", err)
-							}
-						default:
-							klog.V(7).Infof("StreamClient.listen(): msg recv: type %d, len: %d\n", msgType, len(byMsg))
-						}
+				// close the connection
+				c.closeWs(true)
+
+				klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+				return
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/speak/client_stream.go]: I will do this in a subsequent PR. This is tracked by this issue here:
https://github.com/deepgram/deepgram-go-sdk/issues/242

File: pkg/client/live/client.go [hunk 6]
```diff
@@ -298,149 +298,131 @@
...
-					}
-				} else {
-					klog.V(7).Infof("callback is nil: msg recv: type %d, len: %d\n", msgType, len(byMsg))
+				// close the connection
+				c.closeWs(true)
+
+				klog.V(6).Infof("live.listen() LEAVE\n")
+				return
+			}
+		}
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/client/live/client.go:398]: I will do this in a subsequent PR. This is tracked by this issue here:
https://github.com/deepgram/deepgram-go-sdk/issues/242

**Review by SandraRodgers — APPROVED** [review]

**Pull request merged**